python/triton/runtime/jit.py

```diff
@@ -687,6 +687,13 @@ def run(self, *args, grid, warmup, **kwargs):
         # the type and the second parameter is the 'specialization' value.
         bound_args, specialization, options = binder(*args, **kwargs)
 
+        # add a cache field to the kernel specializations for kernel specific
+        # pass pipelines
+        if knobs.runtime.add_stages_inspection_hook is not None:
+            passes = inspect.getsource(knobs.runtime.add_stages_inspection_hook)
+            passes_hash = hashlib.sha256(passes.encode("utf-8")).hexdigest()
+            specialization.append(f'("custom_pipeline", {passes_hash})')
```

Comment:
this code is called for every dispatch, I guess it should be fairly cheap. Did you measure?

Reply:
fair enough, ignore this part then.